File: research-service-py/app/storage/memory_store.py
from __future__ import annotations
from typing import Dict, Optional
from ..models import ResearchBrief


class MemoryStore:
    def __init__(self) -> None:
        self._briefs: Dict[str, ResearchBrief] = {}

    def save_brief(self, brief: ResearchBrief) -> None:
        self._briefs[brief.briefId] = brief

    def get_brief(self, brief_id: str) -> Optional[ResearchBrief]:
        return self._briefs.get(brief_id)


store = MemoryStore()

File: research-service/src/storage/memoryStore.ts
import type { ResearchBrief } from '../types/types';

const briefs = new Map<string, ResearchBrief>();

function saveBrief(brief: ResearchBrief) {
  briefs.set(brief.briefId, brief);
}

function getBrief(id: string): ResearchBrief | undefined {
  return briefs.get(id);
}

export const memoryStore = { saveBrief, getBrief };

